library/common/extensions/stat_sinks/metrics_service/metrics_service_with_response.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/random_generator.h"
+#include "envoy/grpc/async_client.h"
+#include "envoy/local_info/local_info.h"
+#include "envoy/network/connection.h"
+#include "envoy/service/metrics/v3/metrics_service.pb.h"
+#include "envoy/stats/sink.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "library/common/extensions/stat_sinks/metrics_service/metrics_service.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace EnvoyMobileMetricsService {
+
+/**
+ * EnvoyMobile implementation of GrpcMetrics\Streamer
+ */
+class GrpcMetricsEnvoyMobileStreamerImpl
```

Comment:
```suggestion
class EnvoyMobileGrpcMetricsStreamerImpl
```

I'd keep this consistent with the other uses of the prefix EnvoyMobile